### PR TITLE
Calendar

### LINKED ISSUE #42: Research Calendar Implementation
A very open ended issue

Start looking into how we can implement the calendar for V2, including what packages to use for the frontend and how to utilize the google calendar api to export calendar events.

### LINKED ISSUE #68: Event Calendar
Event Calendar

### PR DESCRIPTION
**Pull request by Cubified**
### Administrative Info
[#42 and #68]

### Changes
What changes did you make?
- Added the calendar page
    - Backend routes to handle and store calendar events (including connection between users and events)
       - Backend route to produce an iCal `.ics` file for use with Google Calendar/others
    - View event/add event modals
    - Calendar/scheduler components using [react-simple-scheduler](https://www.npmjs.com/package/@cubedoodl/react-simple-scheduler)
    - Simplified mobile view modeled after Google Calendar's mobile app

### Testing
How did you confirm your changes worked? 
- Manual testing, including usability testing with people who had not seen the site before

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Demos |
| --- |
| main |
| add_event |
| nonadmin |
| mobile |
| anim |

### CODE REVIEW

**Review by PatrickBrown1 — COMMENTED** [review]
Starting to look through this PR in depth. Will note anything I find here

Weird shadow here on view event modal:
image

**Comment by PatrickBrown1** [comment]
Event repeating option should probably be removed from create event modal until implemented (unsure about priority of recurring event implementation)

**Comment by PatrickBrown1** [comment]
Waiting on role PR to merge

File: backend/routes/UserRoutes.js [hunk 1]
```diff
@@ -25,12 +25,14 @@ const upload = multer({
 // returns all users
 router.get("/users", (req, res) =>
   UserModel.find()
+    .populate("events")
```
[the remark this answers is not in the document]
Comment (Cubified) [reply backend/routes/UserRoutes.js:28]: Looking back over this I see it's a waste for this route, there isn't a use case where we need every user with every event they're attending

**Review by aksharans — COMMENTED** [review]
On a first look through, just a couple of things I noticed for now before more testing:
- users with the same name causes issues in the Assign Modal
- the mobile calendar create new event modal was cut off at the top/bottom
- (design suggestion) on mobile view, if there are no events, could have some text saying no events instead of the empty space and blue bar
- (already mentioned) events aren't showing up in a non-admin volunteer's calendar, if they're assigned an event to the user/their roles by an admin

**Review by PatrickBrown1 — APPROVED** [review]

**Comment by aksharans** [comment]
Looks really good! Just a couple of last things I noticed:
- Not sure if this is just on my end, but I was unable to hide individual calendars for both admins and non-admins directly using the checkboxes
  - if I click on the calendar, the role specific calendars and events then seem to disappear/reappear 
- Adding more than a couple guests hides guest names since the scroll bar overlays over them
- longer guest names/relationships would overflow out of the attendees box, so probably just some overflow/ellipses
(for mobile view of calendar, since if I remember correctly, the calendar was mainly meant for desktop view)
- for users who select they are bringing guests, the mobile calendar view event modal glitches out at 600px
- the responses boxes inside the question responses section of the view event modal get arranged out of place at around 645px and 545px

**Review by aksharans — APPROVED** [review]
LGTM

**Pull request merged**